### PR TITLE
Fix:  Add Target_attribute in Variable_t

### PR DESCRIPTION
**Pull request by adit4443ya**
Towards #5697 
And as discussed in this Thread [comment](https://github.com/lfortran/lfortran/pull/5665#issuecomment-2540789755)
Now we have additional attribute 
With this PR I have this output
```clojure
 aditya-trivedi   lfortran    var ≢    cat integration_tests/nullify_01.f90 
program nullify_01
implicit none

   integer, pointer :: p1, p2
   integer, target :: t1 
   
   p1=>t1
   p2=>t1
   p1 = 1
   nullify(p1, p2)
   
end program nullify_01
 aditya-trivedi   lfortran    var ≢    lfortran integration_tests/nullify_01.f90 --show-asr
(TranslationUnit
    (SymbolTable
        1
        {
            nullify_01:
                (Program
                    (SymbolTable
                        2
                        {
                            p1:
                                (Variable
                                    2
                                    p1
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Pointer
                                        (Integer 4)
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.         //This is Target Attribute
                                ),
                            p2:
                                (Variable
                                    2
                                    p2
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Pointer
                                        (Integer 4)
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.        //This is Target Attribute
                                ),
                            t1:
                                (Variable
                                    2
                                    t1
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .true.      //This is Target Attribute
                                )
                        })
                    nullify_01
                    []
                    [(Associate
                        (Var 2 p1)
                        (Var 2 t1)
                    )
                    (Associate
                        (Var 2 p2)
                        (Var 2 t1)
                    )
                    (Assignment
                        (Var 2 p1)
                        (IntegerConstant 1 (Integer 4) Decimal)
                        ()
                    )
                    (Nullify
                        [(Var 2 p1)
                        (Var 2 p2)]
                    )]
                )
        })
    []
)
 aditya-trivedi   lfortran    var ≢   
```

### CODE REVIEW

**Comment by adit4443ya** [comment]
Failure in CI
corresponds to the Check in `TEST_LLVM.cpp`
```clojure
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:421: ERROR: CHECK( LCompilers::pickle(*asr) == "(TranslationUnit (SymbolTable 3 {f: (Function (SymbolTable 4 {f: (Variable 4 f [] ReturnVar () () Default (Integer 4) () Source Public Required .false.)}) f (FunctionType [] (Integer 4) Source Implementation () .false. .false. .false. .false. .false. [] .false.) [] [] [(Assignment (Var 4 f) (IntegerConstant 4 (Integer 4) Decimal) ())] (Var 4 f) Public .false. .false. ())}) [])" ) is NOT correct!
```
and 
```clojure
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:380: ERROR: CHECK( LCompilers::pickle(*asr) == "(TranslationUnit (SymbolTable 1 {f: (Function (SymbolTable 2 {f: (Variable 2 f [] ReturnVar () () Default (Integer 4) () Source Public Required .false.)}) f (FunctionType [] (Integer 4) Source Implementation () .false. .false. .false. .false. .false. [] .false.) [] [] [(Assignment (Var 2 f) (IntegerConstant 5 (Integer 4) Decimal) ())] (Var 2 f) Public .false. .false. ())}) [])" ) is NOT correct!
```

I will wait for review before updating that test

**Comment by Pranavchiku** [comment]
Yes,you need to use `--show-asr --no-indent` for that case and put it inside the string comparison.

**Comment by Pranavchiku** [comment]
You'll need to add `.false.` at appropriate place instead use the approach given in above comment.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -3613,11 +3615,11 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         SetChar variable_dependencies_vec;
                         variable_dependencies_vec.reserve(al, 1);
                         ASRUtils::collect_variable_dependencies(al, variable_dependencies_vec, type, init_expr, value);
-                        ASR::asr_t *v = ASR::make_Variable_t(al, s.loc, current_scope,
+                        ASR::asr_t *v = ASRUtils::make_Variable_t_util(al, s.loc, current_scope,
                                 s2c(al, to_lower(s.m_name)), variable_dependencies_vec.p,
                                 variable_dependencies_vec.size(), s_intent, init_expr, value,
                                 storage_type, type, type_declaration, s_abi, s_access, s_presence,
-                                value_attr);
+                                value_attr, target_attr);
```
Comment (Pranavchiku) [inline src/lfortran/semantics/ast_common_visitor.h:3622]: this is good

File: tests/reference/asr-modules2-98d8120.stdout [hunk 2]
```diff
@@ -50,6 +50,7 @@
                                                     Public
                                                     Required
                                                     .false.
+                                                    .true.
```
Comment (Pranavchiku) [inline tests/reference/asr-modules2-98d8120.stdout:53]: This is good.

**Review by Pranavchiku — APPROVED** [review]
Looks good, just fix the string check and this PR will be good to merge.

**Review by Pranavchiku — APPROVED** [review]

**Review by certik — APPROVED** [review]
I think this is fine. If it turns out we don't want it, we can remove it later.

**Pull request merged**